src/m365/search/commands/externalconnection/externalconnection-get.spec.ts

```diff
@@ -0,0 +1,258 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../../appInsights';
+import auth from '../../../../Auth';
+import { Logger } from '../../../../cli';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { sinonUtil } from '../../../../utils';
+import commands from '../../commands';
+const command: Command = require('./externalconnection-get');
+
+describe(commands.EXTERNALCONNECTION_GET, () => {
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  const externalConnectionGetResponse: any = {
+    configuration: {
+      authorizedAppIds: []
+    },
+    description: 'Test connection that will not do anything',
+    id: 'TestConnectionForCLI',
+    name: 'Test Connection for CLI'
+  };
+
+  const externalConnectionGetResponseWithAppIDs: any = {
+    configuration: {
+      'authorizedAppIds': [
+        '00000000-0000-0000-0000-000000000000',
+        '00000000-0000-0000-0000-000000000001',
+        '00000000-0000-0000-0000-000000000002'
+      ]
+    },
+    description: 'Test connection that will not do anything',
+    id: 'TestConnectionForCLIWithAppIDs',
+    name: 'Test Connection for CLI with App IDs'
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    (command as any).items = [];
+    
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.EXTERNALCONNECTION_GET), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('gets an external connection', (done: any) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/external/connections/TestConnectionForCLI`) {
+        return Promise.resolve(externalConnectionGetResponse);
+      }
+      return Promise.reject('Invalid request');
+    });
+    const options: any = {
+      debug: false,
+      id: 'TestConnectionForCLI',
+      name: 'Test Connection for CLI',
```

Comment:
This test case is incorrect because we're specifying both id and name whereas in the command we expect either one but not both.